stdlib/public/runtime/ReflectionMirror.cpp

```diff
@@ -1092,3 +1092,28 @@ id swift_reflectionMirror_quickLookObject(OpaqueValue *value, const Metadata *T)
 #endif
 
 #endif  // SWIFT_ENABLE_REFLECTION
+
+#include "ImageInspection.h"
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_API
+const char *swift_keypath_dladdr(void *address) {
+    SymbolInfo info;
+    if (lookupSymbol(address, &info) == 0) {
+        return 0;
+    } else {
+        return info.symbolName.get();
+    }
+}
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_API
+const char *swift_keypathSourceString(char *name) {
+    size_t length = strlen(name);
+    std::string mangledName = keypathSourceString(name, length);
+    if (mangledName == "") {
+        return 0;
+    } else {
+        char *rtnValue = new char[mangledName.length() + 1];
+        strcpy(rtnValue, mangledName.c_str());
+        return rtnValue;
+    }
+}
```

Comment:
These should be within the `SWIFT_ENABLE_REFLECTION` condition because we don't want them if reflection is turned off.

Reply:
I can change  it. Fwiw, this part can work without reflection metadata, that's why I left it in. 

My intention was that we would continue to make best effort attempts at producing something useful as long as the requisite data might be available, and that whether this part worked or not would be determined by the IFDEFs in ImageInspection.h. But I guess this still qualifies as reflection, even if it does not use the traditional mechanisms.